### PR TITLE
Add byteutil package

### PR DESCRIPTION
**Pull request by dustinxie**

### CODE REVIEW

File: crypto/README.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
+## CGO Binding Guide
```
Comment (dustinxie) [inline crypto/README.md]: this is the guide for ec283 package
keep it for record

File: hash/hash.go [hunk 2]
```diff
@@ -61,3 +63,21 @@ func BytesToHash160(b []byte) Hash160 {
 	copy(h[20-len(b):], b)
 	return h
 }
+
```
Comment (dustinxie) [inline hash/hash.go]: this can deprecate toHash256() in iotex-core/api/api.go

File: hash/hash.go [hunk 3]
```diff
@@ -61,3 +63,21 @@ func BytesToHash160(b []byte) Hash160 {
...
+// HexStringToHash256 decodes the hex string, then copy byte slice into hash
+func HexStringToHash256(s string) (Hash256, error) {
```
Comment (dustinxie) [inline hash/hash.go]: this can deprecate toHash256() in iotex-core/api/api.go

File: byteutil/byteutil.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2018 IoTeX
```
Comment (lizhefeng) [inline byteutil/byteutil.go]: Change to 2019

File: crypto/cryptosort_test.go [hunk 5]
```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2018 IoTeX
```
Comment (lizhefeng) [inline crypto/cryptosort_test.go]: Ditto

File: crypto/merkle.go [hunk 6]
```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2018 IoTeX
```
Comment (lizhefeng) [inline crypto/merkle.go]: Ditto

File: crypto/merkle_test.go [hunk 7]
```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2018 IoTeX
```
Comment (lizhefeng) [inline crypto/merkle_test.go]: Ditto

**Pull request merged**